### PR TITLE
Scissors - Tammy Ip

### PR DESCRIPTION
**Pull request by tammyip**

### CODE REVIEW

**Comment by tammyip** [comment]
Scissors - Tammy Ip

**Comment by tammyip** [comment]
Accidentally created this.

**Comment by jmaddox19** [comment]
This looks like the Pull Request we asked for. Is there a reason why you deleted it?

File: app/routes.py [hunk 1]
```diff
@@ -1,2 +1,340 @@
...
+from app.models.rental import Rental
+from flask import request, Blueprint, make_response, jsonify
+import datetime
+# from datetime import datetime, timedelta
 
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+def is_valid_data(request_body):
```
Comment (jmaddox19) [inline app/routes.py:14]: Interesting! This doesn't do full validation in that it doesn't ensure the keys and values match what would be expected but it a useful way to DRY up some minimal validation!

File: app/models/rental.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# from .customer import Customer
+# from .video import Video
+from datetime import datetime, timedelta
+
+class Rental(db.Model):
+    rental_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    # a foreign key column refers to the primary key of the other table
+    customer_id = db.Column(db.Integer, db.ForeignKey('customer.customer_id'), nullable=True, default=None) #nullable=True means Null is allowed in this column
+    video_id = db.Column(db.Integer, db.ForeignKey('video.video_id'), nullable=True, default=None)
+    due_date = db.Column(db.DateTime, default=datetime.utcnow() + timedelta(days=7))
```
Comment (jmaddox19) [inline app/models/rental.py:14]: Cool that you added a default value!

File: app/routes.py [hunk 3]
```diff
@@ -1,2 +1,340 @@
...
+    # sort = request.args.get("sort")
+    # page_size = request.args.get("n")
+    # page_number = request.args.get("p")
+
+    # sort_order_dict = { 'name': Customer.name.asc(), 'id': Customer.id.asc() }
+    # sort_order = sort_order_dict[sort]
+
+    # Customer.query.order_by(sort_order).offset(offset).limit(count).all()
+    query_param_value = request.args.get("sort")
+    if query_param_value == "name":
```
Comment (jmaddox19) [inline app/routes.py:36]: Nice!

**Review by jmaddox19 — COMMENTED** [review]
Looks great! Great work here!

**Pull request closed without merging**